posts/2024-09-05-Rust-1.81.0.md

```diff
@@ -0,0 +1,160 @@
+if code runs in an environment without floating point support, you could use
+Clippy to lint on such usage with `#![deny(clippy::float_arithmetic)]`.
+However, if a new developer to the project sees this lint fire, they need to
+look for (hopefully) a comment on the deny explaining why it was added. With
+Rust 1.71, they can be informed directly in the compiler message:
+
+```text
+error: floating-point arithmetic detected
+ --> src/lib.rs:4:5
+  |
+4 |     a + b
+  |     ^^^^^
+  |
+  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#float_arithmetic
+  = note: no hardware float support
+note: the lint level is defined here
+ --> src/lib.rs:1:9
+  |
+1 | #![deny(clippy::float_arithmetic, reason = "no hardware float support")]
+  |         ^^^^^^^^^^^^^^^^^^^^^^^^
+```
+
+### Stabilized APIs
+
+- [`core::error`](https://doc.rust-lang.org/stable/core/error/index.html)
+- [`hint::assert_unchecked`](https://doc.rust-lang.org/stable/core/hint/fn.assert_unchecked.html)
+- [`fs::exists`](https://doc.rust-lang.org/stable/std/fs/fn.exists.html)
+- [`AtomicBool::fetch_not`](https://doc.rust-lang.org/stable/core/sync/atomic/struct.AtomicBool.html#method.fetch_not)
+- [`Duration::abs_diff`](https://doc.rust-lang.org/stable/core/time/struct.Duration.html#method.abs_diff)
+- [`IoSlice::advance`](https://doc.rust-lang.org/stable/std/io/struct.IoSlice.html#method.advance)
+- [`IoSlice::advance_slices`](https://doc.rust-lang.org/stable/std/io/struct.IoSlice.html#method.advance_slices)
+- [`IoSliceMut::advance`](https://doc.rust-lang.org/stable/std/io/struct.IoSliceMut.html#method.advance)
+- [`IoSliceMut::advance_slices`](https://doc.rust-lang.org/stable/std/io/struct.IoSliceMut.html#method.advance_slices)
+- [`PanicHookInfo`](https://doc.rust-lang.org/stable/std/panic/struct.PanicHookInfo.html)
+- [`PanicInfo::message`](https://doc.rust-lang.org/stable/core/panic/struct.PanicInfo.html#method.message)
+- [`PanicMessage`](https://doc.rust-lang.org/stable/core/panic/struct.PanicMessage.html)
+
+These APIs are now stable in const contexts:
+
+- [`char::from_u32_unchecked`](https://doc.rust-lang.org/stable/core/char/fn.from_u32_unchecked.html) (function)
+- [`char::from_u32_unchecked`](https://doc.rust-lang.org/stable/core/primitive.char.html#method.from_u32_unchecked) (method)
+- [`CStr::count_bytes`](https://doc.rust-lang.org/stable/core/ffi/c_str/struct.CStr.html#method.count_bytes)
+- [`CStr::from_ptr`](https://doc.rust-lang.org/stable/core/ffi/c_str/struct.CStr.html#method.from_ptr)
+
+### Compatibility notes
+
+#### Split panic hook and panic handler arguments
+
+We have renamed [`std::panic::PanicInfo`] to [`std::panic::PanicHookInfo`]. The old
+name will continue to work as an alias, but will result in a deprecation
+warning starting in Rust 1.82.0.
+
+`core::panic::PanicInfo` will remain unchanged, however, as this is now a
+*different type*.  cuviper marked this conversation as resolved.
```

Comment:
Copy paste mistake

Reply:
```suggestion
*different type*.
```

Indeed, though a bit of an amusing one to leave in :)